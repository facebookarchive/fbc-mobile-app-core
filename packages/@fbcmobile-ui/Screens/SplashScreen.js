/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 */

'use strict';

import * as React from 'react';
import Spinner from 'react-native-spinkit';
import Text from '@fbcmobile/ui/Components/Core/Text';
import {ApplicationStyles, Colors} from '@fbcmobile/ui/Theme';
import {StyleSheet, View} from 'react-native';

type Props = {
  +text?: ?React.Node,
};

const SplashScreen = (props: Props) => {
  const {text} = props;

  return (
    <View style={styles.container}>
      {text != null && (
        <Text style={styles.textStyle} variant="h6">
          {text}
        </Text>
      )}
      <Spinner size={40} color={Colors.Blue} type={'ThreeBounce'} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    ...ApplicationStyles.screen.container,
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  textStyle: {
    textAlign: 'center',
  },
});

export default SplashScreen;
