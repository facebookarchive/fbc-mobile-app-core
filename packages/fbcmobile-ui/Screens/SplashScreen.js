/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 */

'use strict';

import * as React from 'react';
import Spinner from 'react-native-spinkit';
import {ApplicationStyles, Colors} from '@fbcmobile/ui/Theme';
import {StyleSheet, View} from 'react-native';
import {Text} from '@99xt/first-born';

type Props = {
  +text?: ?React.Node,
};

const SplashScreen = (props: Props) => {
  const {text} = props;

  return (
    <View style={styles.container}>
      {text != null && (
        <Text style={styles.textStyle} size="h6">
          {text}
        </Text>
      )}
      <Spinner size={40} color={Colors.Blue} type={'ThreeBounce'} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    ...ApplicationStyles.screen.container,
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  textStyle: {
    textAlign: 'center',
  },
});

export default SplashScreen;
