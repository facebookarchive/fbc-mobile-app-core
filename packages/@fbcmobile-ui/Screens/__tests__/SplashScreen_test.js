/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 */

'use strict';

import React from 'react';
import SplashScreen from '../SplashScreen';
import renderer from 'react-test-renderer';

describe('SplashScreen Tests', () => {
  test('SplashScreen renders correctly', () => {
    const tree = renderer.create(<SplashScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('SplashScreen renders correctly with text', () => {
    const tree = renderer.create(<SplashScreen text={'testing'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
