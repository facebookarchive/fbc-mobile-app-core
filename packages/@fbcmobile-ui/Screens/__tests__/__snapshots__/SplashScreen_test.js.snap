// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashScreen Tests SplashScreen renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FBFBFB",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RNSpinkit
    color={4279793650}
    size={40}
    style={
      Array [
        Object {
          "height": 40,
          "width": 40,
        },
        undefined,
      ]
    }
    type="ThreeBounce"
  />
</View>
`;

exports[`SplashScreen Tests SplashScreen renders correctly with text 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FBFBFB",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Array [
        Array [
          Object {
            "color": "#484848",
            "fontSize": 15,
            "fontWeight": "500",
            "lineHeight": 20,
          },
          null,
          null,
          Object {
            "textAlign": "left",
          },
        ],
        Object {
          "textAlign": "center",
        },
      ]
    }
  >
    testing
  </Text>
  <RNSpinkit
    color={4279793650}
    size={40}
    style={
      Array [
        Object {
          "height": 40,
          "width": 40,
        },
        undefined,
      ]
    }
    type="ThreeBounce"
  />
</View>
`;
